internal/datastore/postgres/gc.go

```diff
@@ -71,16 +71,18 @@ func (pgd *pgDatastore) TxIDBefore(ctx context.Context, before time.Time) (datas
 		return datastore.NoRevision, err
 	}
 
-	return revisionFromTransaction(value), nil
+	return postgresRevision{value, xmin}, nil
 }
 
 func (pgd *pgDatastore) DeleteBeforeTx(ctx context.Context, txID datastore.Revision) (removed common.DeletionCounts, err error) {
+	revision := txID.(postgresRevision)
+
 	// Delete any relationship rows with deleted_transaction <= the transaction ID.
 	removed.Relationships, err = pgd.batchDelete(
 		ctx,
 		tableTuple,
 		relationTuplePKCols,
-		sq.LtOrEq{colDeletedXid: transactionFromRevision(txID)},
+		sq.LtOrEq{colDeletedXid: revision.tx},
```

Comment:
do we need to query against the xmins too?

I don't know that an application should be reasoning about liveness at the GC threshold, but I think this means you could send in a revision (close to the GC threshold) twice in a row and get different answers.

Ideally what you'd want is just one answer followed by an "revision out of bounds" error, right?